text/0000-rubygems.org-sigstore-bundles.md

```diff
@@ -0,0 +1,57 @@
+- Feature Name: RubyGems.org accepts Sigstore Bundles with Gem Pushes
+- Start Date: 2024-10-24
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+RubyGems.org will add a new variant to the gem push endpoint that allows gems authors to include sigstore bundles with their gem pushes.
+RubyGems.org will verify that the sigstore bundle is valid and signed by a trusted publisher configured for the gem.
```

Comment:
To confirm, will trusted publishing be a requirement to accept Sigstore bundles, as in you won't accept signatures generated by a non-CI workflow identity?

Reply:
To start, I plan on limiting to trusted publishing, since it gives us a solid basis for an identity to verify against. I believe this is the approach that pypi is taking. Going forward though, I see no reason why rubygems.org fundamentally needs to restrict what attestations we accept, given they are coming from an actor that has permissions to publish to the gem

Reply:
One benefit for only accepting trusted publishing identities is if you want to provide a default verification policy, as you don't have to worry about exposing non-public maintainer details (except for dealing with private repos).